src/ripple/protocol/impl/TER.cpp

```diff
@@ -92,6 +92,7 @@ transResults()
         MAKE_ERROR(tecINSUFFICIENT_FUNDS,            "Not enough funds available to complete requested transaction."),
         MAKE_ERROR(tecOBJECT_NOT_FOUND,              "A requested object could not be located."),
         MAKE_ERROR(tecINSUFFICIENT_PAYMENT,          "The payment is not sufficient."),
+        MAKE_ERROR(tecEMPTY_DID,                     "The DID object cannot be made to be empty."),
```

Comment:
Suggestion: I think it would be useful to say "Atleast URI or Data field must be non-empty in a DID object"